src/TTextEdit.cpp

```diff
@@ -1421,6 +1426,83 @@ void TTextEdit::slot_copySelectionToClipboardHTML()
     return;
 }
 
+void TTextEdit::slot_copySelectionToClipboardImage()
+{
+    //this->grab().save(QStringLiteral("/tmp/image.png"));
+
+    //    const QRect& rect = e->rect();
+
+    // mPA QPoint where selection started
+    // mPB QPoint where selection ended
+
+    // if selection was made backwards swap
+    // right to left
+    if ((mPA.y() == mPB.y()) && (mPA.x() > mPB.x())) {
+        swap(mPA, mPB);
+    }
+    // down to up
+    if (mPA.y() > mPB.y()) {
+        swap(mPA, mPB);
+    }
+
+    if (mFontWidth <= 0 || mFontHeight <= 0) {
+        return;
+    }
+
+    if (mScreenHeight <= 0 || mScreenWidth <= 0) {
+        mScreenHeight = height() / mFontHeight;
+        mScreenWidth = 100;
+        if (mScreenHeight <= 0 || mScreenWidth <= 0) {
+            return;
+        }
+    }
+
+    auto height = (mPB.y() - mPA.y()+1)*mFontHeight;
+
+    // find the biggest width of text we need to work with
+    int characterWidth = 0;
+    for (int y = mPA.y(), total = mPB.y() + 1; y <= total; ++y) {
+        auto lineWidth = static_cast<int>(mpBuffer->buffer[y].size());
+        characterWidth = lineWidth > characterWidth ? lineWidth : characterWidth;
+    }
+
+
+    auto width = characterWidth*mFontWidth;
+
+    if (width >= 1'000'000) {
+        width = 1'000'000;
+    }
+    if (height >= 1'000'000) {
+        height = 1'000'000;
+    }
+
+    auto rect = QRect(mPA.x(), mPA.y(), width, height);
+    qDebug() << "rect" << rect;
+
+    auto pix = QPixmap(width, height);
+
+    QPainter painter(&pix);
+    if (!painter.isActive()) {
+        return;
+    }
+
+    // deselect to prevent inverted colours in image
```

Comment:
Not the ideal case - the other copy routines did not deselect the selection IIRC (or was that only in my stalled TBuffer refactor/clean-up/polishing/extending PR....) - but I guess the only way around it would be a revised `drawForeground` that ignored the *selection* Fore/Back ground inversion so the painted image does not carry that effect...  :slightly_frowning_face:

Reply:
Done